pkg/documentlog/interface.go

```diff
@@ -17,14 +17,14 @@ type DocumentLog interface {
 }
 
 type DocumentQueue interface {
-	Get() *model.Document
+	Get() model.Document
 }
 
 type documentQueue struct {
 	documentType string
-	c            chan *model.Document
+	c            chan model.Document
 }
 
-func (q documentQueue) Get() *model.Document {
+func (q documentQueue) Get() model.Document {
```

Comment:
Docs, since it's a blocking call without a timeout.

Reply:
We're missing `Context` here (so it can be cancelled/timed out), adding it.